aiohttp/web_app.py

```diff
@@ -115,7 +122,8 @@ def __init__(
         # initialized on freezing
         self._run_middlewares = None  # type: Optional[bool]
 
-        self._state = {}  # type: Dict[str, Any]
+        # We cheat here slightly, to allow users to type hint their apps more easily.
+        self._state: _T = {}  # type: ignore[assignment]
```

Comment:
I agree this is probably the right thing to do. I tried to take a look at ways this could be fixed but couldn't come up with any. I was thinking something along the lines of overloading `__init__` but that doesn't seem to be the right approach?

Reply:
We're literally ignoring the type and assigning an empty dict. I don't think there's any way to achieve this without cheating the typing or doing something worse. I want to avoid asking the user to pass in a pre-made dict (which would be required for technically correct typing), as most of the dict values should typically be created in an app context.